### ISSUE TITLE
Builld error with mingw gcc 5.5

### ISSUE THREAD

**Issue by Benau** [kcat/openal-soft#690]
```
/data/game/dependencies/src/openal/common/aloptional.h:316:26: error: cannot convert 'al::optional<float>' to 'bool' in initialization
     { return bool{*this} ? std::move(**this) : static_cast<T>(std::forward<U>(defval)); }
```
Do you think
```
 sed -i 's/bool{\*this}/has_value()/g' common/aloptional.h
```
make sense?

**Comment by kcat** [comment]
It's essentially defined as cppreference defines it: https://en.cppreference.com/w/cpp/utility/optional/value_or The main difference being the use of `bool{*this}` instead of `bool(*this)`. Though I don't see how `has_value()` would be a problem.

**Comment by Benau** [comment]
using ```bool(*this)``` compiles too, do you think you can commit the fix?
```patch
diff --git a/common/aloptional.h b/common/aloptional.h
index 6180d161..6de16799 100644
--- a/common/aloptional.h
+++ b/common/aloptional.h
@@ -310,10 +310,10 @@ public:
 
     template<typename U>
     constexpr T value_or(U&& defval) const&
-    { return bool{*this} ? **this : static_cast<T>(std::forward<U>(defval)); }
+    { return bool(*this) ? **this : static_cast<T>(std::forward<U>(defval)); }
     template<typename U>
     constexpr T value_or(U&& defval) &&
-    { return bool{*this} ? std::move(**this) : static_cast<T>(std::forward<U>(defval)); }
+    { return bool(*this) ? std::move(**this) : static_cast<T>(std::forward<U>(defval)); }
 
     template<typename ...Args>
     constexpr T& emplace(Args&& ...args)
```

**Comment by kcat** [comment]
Fixed with commit 7c33c1b2e677acfc9977d6e90dbfc72674f091ca.
